src/runtime_src/core/tools/common/tests/TestTCTAllColumn.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "TestTCTAllColumn.h"
+#include "tools/common/XBUtilities.h"
+#include "tools/common/BusyBar.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_kernel.h"
+namespace XBU = XBUtilities;
+
+// System - Include Files
+#include <fstream>
+#include <filesystem>
+
+static constexpr size_t host_app = 1; //opcode
+static constexpr size_t buffer_size = 4;
+static constexpr size_t word_count = buffer_size/4;
+static constexpr int itr_count = 20000;
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+TestTCTAllColumn::TestTCTAllColumn()
+  : TestRunner("tct-all-col", 
+                "Measure average TCT processing time for all columns",
+                "validate.xclbin")
+                {
+                  m_dpu_name = "tct_4col.txt";
+                }
+
+namespace {
+
+// Copy values from text files into buff, expecting values are ascii encoded hex
+static void 
+init_instr_buf(xrt::bo &bo_instr, const std::string& dpu_file) {
+  std::ifstream dpu_stream(dpu_file);
+  if (!dpu_stream.is_open()) {
+    throw std::runtime_error(boost::str(boost::format("Failed to open %s for reading") % dpu_file));
+  }
+
+  auto instr = bo_instr.map<int*>();
+  std::string line;
+  while (std::getline(dpu_stream, line)) {
+    if (line.at(0) == '#') {
+      continue;
+    }
+    std::stringstream ss(line);
+    unsigned int word = 0;
+    ss >> std::hex >> word;
+    *(instr++) = word;
+  }
+}
+
+static size_t 
+get_instr_size(const std::string& dpu_file) {
+  std::ifstream file(dpu_file);
+  if (!file.is_open()) {
+    throw std::runtime_error(boost::str(boost::format("Failed to open %s for reading") % dpu_file));
+  }
+  size_t size = 0;
+  std::string line;
+  while (std::getline(file, line)) {
+    if (line.at(0) != '#') {
+      size++;
+    }
+  }
+  if (size == 0) {
+    throw std::runtime_error("Invalid DPU instruction length");
+  }
+  return size;
+}
+
+} //anonymous namespace
+
+boost::property_tree::ptree
+TestTCTAllColumn::run(std::shared_ptr<xrt_core::device> dev)
+{
+  boost::property_tree::ptree ptree = get_test_header();
+
+  // workaround: can't rename files when copying to driver store
+  // so need to name the files as _phx and _stx
+  // will revisit this after the current release
+  auto device_id = xrt_core::query::pcie_device::to_string(xrt_core::device_query<xrt_core::query::pcie_device>(dev));
```

Comment:
Below conditions should be under WIN32 flags. Please fix it in next PR. 
Reference from https://github.com/Xilinx/XRT/pull/7931